Content.Server/Access/Systems/ActivatableUIRequiresAccessSystem.cs

```diff
@@ -0,0 +1,34 @@
+using Content.Server.Lock.Components;
+using Content.Server.Popups;
+using Content.Server.UserInterface;
+using Content.Server.Access.Components;
+using Content.Shared.Access.Components;
+using Content.Shared.Access.Systems;
+
+namespace Content.Server.Access.Systems;
+public sealed class ActivatableUIRequiresAccessSystem : EntitySystem
+{
+    [Dependency] private readonly AccessReaderSystem _access = default!;
+    [Dependency] private readonly ActivatableUISystem _activatableUI = default!;
+    [Dependency] private readonly PopupSystem _popupSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<ActivatableUIRequiresAccessComponent, ActivatableUIOpenAttemptEvent>(OnUIOpenAttempt);
+    }
+
+    private void OnUIOpenAttempt(Entity<ActivatableUIRequiresAccessComponent> component, ref ActivatableUIOpenAttemptEvent args)
+    {
+        if (args.Cancelled)
+            return;
+
+        if (TryComp<AccessReaderComponent>(component, out var accessReader) && !_access.IsAllowed(args.User, component, accessReader))
```

Comment:
IsAllowed already does TryComp interally dont need it
just do if (!_access.IsAllowed(args.User, component))

also name the arg ent not component, the comp is .Comp